test/unit/util.test.js

```diff
@@ -36,6 +36,38 @@ describe('UTILITY FUNCTION TESTS ', function () {
       done();
     });
 
+    it('should resolve circular structures', function(done) {
+      let schema = {
+          '$ref': '#/components/schemas/a'
+        },
+        components = {
+          schemas: {
+            'a': {
+              'type': 'array',
+              'items': {
+                '$ref': '#/components/schemas/b'
+              }
+            },
+            'b': {
+              'type': 'object',
+              'properties': {
+                'c': {
+                  '$ref': '#/components/schemas/a'
+                }
+              }
+            }
+          }
+        },
+        bodyType = 'REQUEST',
+
+        result = Utils.safeSchemaFaker(schema, bodyType, components);
+
+      expect(result).to.not.equal(null);
+      expect(_.isEqual(result[0].c[0].c[0].c[0].c[0].c, {
```

Comment:
overly complicated
`expect(r.0.c.0.c.value).to.equal('Error--too many levels')`